src/bin/pg_dump/pg_dump.c

```diff
@@ -2223,6 +2241,19 @@ dumpTableData_insert(Archive *fout, const void *dcontext)
 						appendPQExpBufferStr(insertStmt,
 											 fmtId(PQfname(res, field)));
 					}
+
+					/*
+					 * Replace 'rolname' column we added earlier with 'orig_loginname' in
+					 * insert statement's column list.
+					 */
+					if (!found && isBabelfishDatabase(fout) && tbinfo->dobj.namespace &&
+						pg_strcasecmp(tbinfo->dobj.namespace->dobj.name, "sys") == 0 &&
+						pg_strcasecmp(tbinfo->dobj.name, "babelfish_authid_login_ext") == 0)
+					{
+						insertStmt->len -= strlen(fmtId("rolname"));
+						insertStmt->data[insertStmt->len] = '\0';
```

Comment:
This has an implicit assumption that `rolname` is the last column. Just to be safe and clear, could you add an assertion?